### PR TITLE
Fill in ancestry/migration events section

### PR DESCRIPTION
**Pull request by awohns**
Here's what I've written for `record_migrations` in the ancestry docs, using the examples I showed in #1272

### CODE REVIEW

File: docs/ancestry.md [hunk 1]
```diff
@@ -961,11 +961,150 @@ section for more details.
...
+        time = []
+        while i != -1:
+            migs = np.where(mig.node == i)[0]
+            for cur_mig in migs:
+                cur_mig = mig[cur_mig]
+                loc.append(cur_mig.dest + 0.5)
+                time.append(cur_mig.time)
+            i = tree.parent(i)
+        plt.plot(loc, time, linestyle=linestyle, color="black")
+    for i in range(10):
```
Comment (jeromekelleher) [inline docs/ancestry.md:1033]: Are these 10s really ``N``?

File: docs/ancestry.md [hunk 2]
```diff
@@ -961,11 +961,150 @@ section for more details.
...
+    import numpy as np
+
+    cmap = matplotlib.cm.get_cmap('viridis')
+    tree = ts.first()
+    mig_color_dict = {}
+    for i, linestyle in zip(ts.samples()[::2], ["dotted", "dashdot"]):
+        mig = ts.tables.migrations
+        loc = []
+        time = []
+        while i != -1:
```
Comment (jeromekelleher) [inline docs/ancestry.md:1025]: -1 -> tskit.NULL

Comment (jeromekelleher) [reply docs/ancestry.md:1025]: It's not helpful to use ``i`` as a node id here and also as a generic index variable just below. We usually use ``u`` or ``v`` as node variables.

**Review by jeromekelleher — COMMENTED** [review]
Looks great, thanks @awohns! I think it's a bit more elaborate than it needs to be though: we're trying to explain what migration records are and how to use them, not necessarily explain anything about random walks on the tree topology. I think we can simplify things a bit and aim to have all the code visible.

**Comment by awohns** [comment]
Thanks @jeromekelleher. I made the suggested changes and removed the references to "random walks". Let me know if you think anything else should be changed.

**Review by jeromekelleher — APPROVED** [review]
LGTM, thanks.

**Pull request merged**